### PR TITLE
explicit ValueError for built-in variations

### PR DESCRIPTION
**Pull request by Akshaysehgal2005**
The issue arises in the specific case when a string identifier is passed to the "metrics" parameter for model.compile(), which is callable but aimed at one of these metrics: `['accuracy', 'acc', 'crossentropy', 'ce']`. 

**Example : "Accuracy" instead of "accuracy" :**

It seems that when "Accuracy" is passed as a string (instead of "accuracy" lowercase) to the metric parameter, the compile deserializes it to get `tf.keras.metrics.Accuracy` directly instead of fetching `tf.keras.metrics.MeanMetricWrapper` as defined in `_get_metric_object()` from compile_utils. Therefore, the accuracy metric is still visible during Epochs but incorrectly calculated (always equal to 0) without throwing any warnings/errors. The ideal state should be to throw `ValueError` instead of failing silently.

I have added an explicit condition that checks if `str.lower()` is present in the list `['accuracy', 'acc', 'crossentropy', 'ce']`. **This list is explicitly conditionally handled** in the `_get_metric_object()` from the [tf.python.keras.engine.compile_utils.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/engine/compile_utils.py/#L441-L444).

-----------

A detailed description of the issue is available [here](https://github.com/tensorflow/tensorflow/issues/46436).

### CODE REVIEW

**Comment by Akshaysehgal2005** [comment]
@googlebot I signed it!

**Comment by Akshaysehgal2005** [comment]
Contributors, requesting a code review thanks!

**Comment by Akshaysehgal2005** [comment]
Hi, Is any more information needed from my side?

**Comment by tomerk** [comment]
Sorry, forgot to press enter when I wrote up a response previously!  Thanks for bringing this to our attention! We're going to try taking a closer at this since it highlights some rough edges around our serialization mechanisms and it's not immediately obvious this PR is the right thing to do.

**Comment by Akshaysehgal2005** [comment]
@tomerk, thanks! I have detailed out the trace for the bug in this issue https://github.com/tensorflow/tensorflow/issues/46436.. This describes why and how the bug is occurring and I have primarily made a PR for handling the issue. I would love to discuss and help fix it at a deeper level and update the PR if required. Do feel free to involve me, would love to contribute! Thanks!

**Comment by gbaned** [comment]
@Akshaysehgal2005 Can you please resolve conflicts? Thanks!

**Comment by gbaned** [comment]
@Akshaysehgal2005 Any update on this PR? Please. Thanks!

**Comment by Akshaysehgal2005** [comment]
Apologies, will resolve it in a day, has been a bit crazy in my country due to the Covid situation. Apologies again @gbaned

**Comment by deeb02** [comment]
@Akshaysehgal2005 , as was mentioned in the discussion earlier, we are investigating deeper solutions to this bug at the moment. If you are interested in a deeper fix at the root cause (serialization/deserialization) please start a new PR on https://github.com/keras-team/keras
